package/com.unity.formats.usd/Runtime/Scripts/IO/SanitizedSamples.cs

```diff
@@ -486,44 +492,67 @@ static void FlattenPrimvar(Primvar<object> primvar)
             }
         }
 
-        internal static void TriangulateAttributes<T>(ref T[] values, int[] faceVertexCount, bool changeHandedness)
+        internal static void TriangulateAttributes<T>(ref T[] values, int[] faceVertexCounts, bool changeHandedness)
         {
-            var newValues = new List<T>();
-            var last = 0;
-            for (var i = 0; i < faceVertexCount.Length; i++)
+            int newFacesCount = 0;
+            for (int i = 0; i < faceVertexCounts.Length; i++)
```

Comment:
a little nitpick but can we perhaps change this to a foreach loop for better readability?

Reply:
I guess that's a matter of preference, personally I find a for loop more readable 😅 Maybe that's just me though, if you prefer foreach I will change it

Reply:
thank you, but as you said its a matter of preference so its okay, i also see that you used for loops instead of foreach for the loops below, so i think its better to be consistent

Plus I once read that foreach loop was faster than for-loops, but after checking it again, it seems that i read it wrong and there's no real difference in real-world usage